src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -863,7 +863,13 @@ private void DoCloseHelper()
                 }
 
                 if ((hostRunspace == null) || (this == hostRunspace))
-                {
+                {                   
+                    // We should not close transcripting if we are not under default runspace.
```

Comment:
I was confused by the code and the comment as I misread the comment initially.  I think you should change the comment to avoid a double negative.

```
We should close transcripting only if we are running in default runspace
```